ethereum/p2p/src/main/java/tech/pegasys/pantheon/ethereum/p2p/peers/cache/PersistentJsonPeerCache.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.pantheon.ethereum.p2p.peers.cache;
+
+import tech.pegasys.pantheon.util.enode.EnodeURL;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import io.vertx.core.json.DecodeException;
+import io.vertx.core.json.JsonArray;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+public class PersistentJsonPeerCache extends PersistentPeerCache {
+
+  private static final Logger LOG = LogManager.getLogger();
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
```

Comment:
JDK constant of `StandardCharsets.UTF_8` would be a nicer fit

Reply:
Done.